common/compositor/va/varenderer.cpp

```diff
@@ -251,15 +191,13 @@ bool VARenderer::Draw(const MediaState& state, NativeSurface* surface) {
     return false;
   }
 
-  ScopedVAConfigID va_config(va_display_);
-  if (!va_config.CreateConfig(rt_format)) {
-    return false;
-  }
-
-  ScopedVAContextID va_context(va_display_);
-  if (!va_context.CreateContext(va_config, dest_width, dest_height,
-                                surface_out)) {
-    return false;
+  if (va_context_ == VA_INVALID_ID || render_target_format_ != rt_format) {
+    DTRACE("to create VA context\n");
+    render_target_format_ = rt_format;
+    if (!CreateContext()) {
+      ETRACE("Create VA context failed\n");
+      return false;
+    }
   }
 
   VAProcPipelineParameterBuffer param;
```

Comment:
You should also be able to set this capabilities only when context is created.

Reply:
it won't run to here if context is not created.